docs/source/guide/Modules.lrad

```diff
@@ -0,0 +1,70 @@
+# Modules
+
+Modules are dicts with some metadata about a module, and an `:env` key
+containing a radicle environment.
+
+To create a module, you can use the `module` special form. All modules must
+start with a *module declaration* which provides the module with a name,
+documentation and an exports vector. Module declarations are evaluated, so you
+must quote any symbols.
+
+    (module
+      {:module  'my-module
+       :doc     "Just for showing off modules."
+       :exports '[foo bar]}
+
+      (def foo-helper
+        (fn [x] (+ x 1)))
+
+      (def foo
+        (fn [x] (foo-helper (foo-helper x))))
+
+      (def bar
+        (fn [x] (foo (+ x 2)))))
```

Comment:
Could you explain how atoms that are not defined in the module are resolved?

Reply:
I’d actually be very interested in having parameterized modules.
~~~clojure
(module
  {:name 'foo
   :parameters {:base 0}
  }
  (def add (fn [x] (+ base x))
)

(import foo-default foo)
(foo-default/add 1) ;; => 1

(import foo-custom foo {:base 3})
(foo-custom/add 1) ;; => 4

Reply:
That's mostly achievable already just using lambdas:

```clojure
(def mk-foo
  (fn [base]
    (module
     {:module 'foo
      :doc "foo"
      :exports ['add]}
     (def add (fn [x] (+ base x))))))
```

and then:

```
rad> (import (mk-foo 0))
:ok
rad> (foo/add 0)
0
rad> (import (mk-foo 3))
:ok
rad> (foo/add 0)
3
```